src/main/ank/1.6-simple-algebraic-data-types.md

```diff
@@ -0,0 +1,328 @@
+```Haskell
+swap :: (a, b) -> (b, a)
+swap (x, y) = (y, x)
+```
+```kotlin
+val swap : (Pair<A, B>) -> Pair<B, A> = 
+    {it.second to it.first}
+```
+................
+```Haskell
+((a, b), c)
+```
+```kotlin
+((a, b), c)
+```
+................
+```Haskell
+(a, (b, c))
+```
+```kotlin
+(a, (b, c))
+```
+................
+```Haskell
+alpha :: ((a, b), c) -> (a, (b, c))
+alpha ((x, y), z) = (x, (y, z))
+```
+```kotlin
+val alpha: (Pair<Pair<A, B>, C>) -> Pair<A, Pair<B, C>> = 
+   { it.first.first to (it.first.second to it.second)}
+```
+................
+```Haskell
+alpha_inv :: (a, (b, c)) -> ((a, b), c)
+alpha_inv (x, (y, z)) = ((x, y), z)
+```
+```kotlin
+val alpha_inv: (Pair<A, Pair<B, C>>) -> Pair<Pair<A, B>, C>  =
```

Comment:
Since we have to use TupleN let's just use Tuple from arrow for all cases and not Pair so there is no chance of confusion to readers